public/app/features/alerting/unified/components/rule-editor/AlertRuleForm.tsx

```diff
@@ -247,7 +290,13 @@ export const AlertRuleForm: FC<Props> = ({ existing, prefill }) => {
           <CustomScrollbar autoHeightMin="100%" hideHorizontalTrack={true}>
             <div className={styles.contentInner}>
               <AlertRuleNameInput />
-              <QueryAndExpressionsStep editingExistingRule={!!existing} onDataChange={checkAlertCondition} />
+              <QueryAndExpressionsStep
+                editingExistingRule={!!existing}
+                prefill={!!prefill}
+                onDataChange={checkAlertCondition}
+                lazyDefaultQueries={defaultDsAndQueries.queries}
```

Comment:
The prop is named `lazy`, but we render it only if there are some `queries` in the `defaultDsAndQueries` object. So what does `lazy` means in this context?

Reply:
Maybe we could rename it to asycDefaultQueries => this queries array is an async value. Once we have a value in here means we can render.